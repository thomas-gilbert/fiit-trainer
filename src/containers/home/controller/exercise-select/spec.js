import React from 'react';
import { ExerciseSelect } from './index.js';

describe('ExerciseSelect Component', () => {
  it('renders correctly', () => {
    const props = {
      defaultValue: 'Please select',
      addExerciseToList: jest.fn(),
    };
    const component = global.shallow(
      <ExerciseSelect {...props} />
    );

    expect(component).toMatchSnapshot();
  });
});
