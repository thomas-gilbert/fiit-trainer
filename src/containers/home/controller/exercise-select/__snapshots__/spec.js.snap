// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExerciseSelect Component renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <Select
    addLabelText="Add \\"{label}\\"?"
    arrowRenderer={[Function]}
    autosize={true}
    backspaceRemoves={true}
    backspaceToRemoveMessage="Press backspace to remove {label}"
    clearAllText="Clear all"
    clearRenderer={[Function]}
    clearValueText="Clear value"
    clearable={true}
    closeOnSelect={true}
    deleteRemoves={true}
    delimiter=","
    disabled={false}
    escapeClearsValue={true}
    filterOptions={[Function]}
    ignoreAccents={true}
    ignoreCase={true}
    inputProps={Object {}}
    isLoading={false}
    joinValues={false}
    labelKey="label"
    matchPos="any"
    matchProp="any"
    menuBuffer={0}
    menuRenderer={[Function]}
    multi={false}
    noResultsText="No results found"
    onBlurResetsInput={true}
    onChange={[Function]}
    onCloseResetsInput={true}
    onSelectResetsInput={true}
    openOnClick={true}
    optionComponent={[Function]}
    options={
      Array [
        Object {
          "label": "Please select",
          "value": "Please select",
        },
      ]
    }
    pageSize={5}
    placeholder="Select..."
    required={false}
    scrollMenuIntoView={true}
    searchable={true}
    simpleValue={false}
    tabSelectsValue={true}
    value="Please select"
    valueComponent={[Function]}
    valueKey="value"
  />
  <input
    type="submit"
    value="Add"
  />
</form>
`;
